### PR TITLE
4.X: Fix broken PNG behavior on OSX

### LINKED ISSUE #127: OpenCV looking for png.h in wrong location
Appears to want to find `png.h` in `${PREFIX}/include/libpng/png.h`, but it is in `${PREFIX}/include/libpng16/png.h` instead. We either should symlink this directory in `libpng` or add a hack to OpenCV to do this.

```
-- Looking for /Users/travis/miniconda3/conda-bld/opencv_1536446791779/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/include/libpng/png.h
-- Looking for /Users/travis/miniconda3/conda-bld/opencv_1536446791779/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/include/libpng/png.h - not found
```

ref: https://travis-ci.org/conda-forge/opencv-feedstock/jobs/426208605#L784

### PR DESCRIPTION
**Pull request by hajapy**
Closes #127

Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hajapy** [comment]
@hmaarrfk I recently discovered that images can't be opened in opencv on os x since 3.4.7 (3.4.4 was last known good version). Opened this PR to demonstrate. Seems like a libpng version mismatch. Do you know what the issue might be?

**Comment by hajapy** [comment]
Wonder if this is related to https://github.com/conda-forge/opencv-feedstock/issues/127

**Comment by hmaarrfk** [comment]
Nice find

**Comment by hmaarrfk** [comment]
There are two builds of 3.4.4, can you confirm which one worked for you?

File: recipe/build.sh [hunk 1]
```diff
@@ -141,7 +142,9 @@ cmake -LAH -G "Ninja"                                                     \
     -DOPENCV_SKIP_PYTHON_LOADER=1                                         \
     -DZLIB_INCLUDE_DIR=${PREFIX}/include                                  \
     -DZLIB_LIBRARY_RELEASE=${PREFIX}/lib/libz${SHLIB_EXT}                 \
+    -DPNG_PNG_INCLUDE_DIR=${PREFIX}/include                               \
```
Comment (hmaarrfk) [inline recipe/build.sh]: I think this is the important change.

**Comment by hmaarrfk** [comment]
seems to now have found the correct library. CMAKE is just crazy for png, calls it `png_png`

**Comment by hmaarrfk** [comment]
they do some weird stuff where they force search in `include/libpng` while the file we need is in `include/libpng16`.

**Comment by hmaarrfk** [comment]
homebrew doesn't do anything specifal. but maybe their stuff is broken too
https://github.com/Homebrew/homebrew-core/blob/master/Formula/opencv.rb

**Comment by hajapy** [comment]
The working 3.4.4 env had this opencv in it: 
```
opencv                    3.4.4           py37_blas_openblashe2643be_1205  [blas_openblas]  conda-forge
```

Created yesterday with `conda create -n opencvtest -c conda-forge opencv=3.4.4`

**Comment by hmaarrfk** [comment]
I think it has to do with CMake+FindPNG that looks for libpng instead of libpng16

**Comment by hmaarrfk** [comment]
THere is a build in this history that works. I'm just trying to find the minimum set of flags that works at this point. Thank you very much for recreating the issue in a reproducible way.

Should we add the test files in the source with hashes? or leave that to a future cleanup?

**Comment by hmaarrfk** [comment]
@conda-forge/libpng @conda-forge/png

Any idea what the best way to address findpng on Mac is?

It seems this is a bug between png and cmake

**Comment by hajapy** [comment]
> Should we add the test files in the source with hashes? or leave that to a future cleanup?

Let’s improve the test now so we can back port both the fix and sync the test case in one round. I think just checking those files into the recipes dir would work. Thoughts?

**Comment by hmaarrfk** [comment]
I already backported the fix to the 3.4.5, but I guess we can rebase and clean things up a little.

I was using both branches to iterate faster.

I would prefer not putting large images in this repo. Asking conda to manager and cache them seems like a more preferrable solution to me.

**Comment by hajapy** [comment]
Up to you, but FYI the image is tiny and the video file is < 2 MB.

**Comment by hmaarrfk** [comment]
haha, yeah, lets just include the PNG file, video file can stay for a future PR.

**Comment by hmaarrfk** [comment]
I think as the owner of the Repo you should upload or cleanup as you see fit. I can't upload things using the github firefox interface, and too lazy to clone your repo on my computer ;).

Feel free to squash.

**Comment by hajapy** [comment]
How does this look? Still need to bump the build num, can cleanup commits as well.

**Comment by hmaarrfk** [comment]
I think this is good. ripgrep seems like icing on the cake, can probably be removed and moved into the conda-image if we feel it is necessary.

**Comment by hajapy** [comment]
All cleaned up now. First commit shows the broken test case, second commit includes your symlink and cmake flag fix + the build number being bumped.

**Comment by hajapy** [comment]
@conda-forge-admin please rerender

**Comment by hajapy** [comment]
Also, for reference it wasn't just pngs that failed to load. I couldn't load jpgs either. bmp and tiff seem fine.

**Comment by hmaarrfk** [comment]
Do we have a test for jpegs now?

**Comment by hmaarrfk** [comment]
merging this for now. lets fix jpeg in a different pr

**Pull request merged**